src/result.js

```diff
@@ -40,141 +40,32 @@ var {
-                            if (oneres[j].Djeeta[sortkey] >= minSortKey[j]) {
-                                // Add to eleventh
-                                res[j].push({data: oneres[j], armNumbers: combinations[i]});
-
-                                // Erase what matches minSortkey [j] up to the 10th
-                                var spliceid = -1;
-                                for (var k = 0; k < 10; k = (k + 1) | 0) {
-                                    if (res[j][k].data.Djeeta[sortkey] == minSortKey[j]) {
-                                        spliceid = k
-                                    }
-                                }
-                                res[j].splice(spliceid, 1);
-                                minSortKey[j] = -1;
-
-                                // Since it became an array of ten, once again calculate the minimum value
-                                for (var k = 0; k < 10; k = (k + 1) | 0) {
-                                    if (minSortKey[j] < 0 || minSortKey[j] > res[j][k].data.Djeeta[sortkey]) {
-                                        minSortKey[j] = res[j][k].data.Djeeta[sortkey]
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
-                initializeTotals(totals)
-            }
-            // At this point, summonres should be an array of "array of associative arrays of result data corresponding to each summon"
-            for (var i = 0; i < summon.length; i++) {
-                if (sortkey == "ATKandHP") {
-                    res[i].sort(function (a, b) {
-                        if ((a.data.Djeeta.displayAttack + a.data.Djeeta.displayHP) < (b.data.Djeeta.displayAttack + b.data.Djeeta.displayHP)) return 1;
-                        if ((a.data.Djeeta.displayAttack + a.data.Djeeta.displayHP) > (b.data.Djeeta.displayAttack + b.data.Djeeta.displayHP)) return -1;
-                        return 0;
-                    });
-                } else {
-                    res[i].sort(function (a, b) {
-                        if (a["data"]["Djeeta"][sortkey] < b["data"]["Djeeta"][sortkey]) return 1;
-                        if (a["data"]["Djeeta"][sortkey] > b["data"]["Djeeta"][sortkey]) return -1;
-                        return 0;
-                    });
-                }
-                while (res[i].length > 10) {
-                    res[i].pop();
-                }
-            }
-
-            return {summon: summon, result: res, sortkeyname: sortkeyname, totalItr: totalItr}
-        } else {
-            return {summon: summon, result: []}
-        }
+    calculateResult: function ({ profile, armlist, summon, chara, sortKey }) {
+      const { previousArmlist, previousCombinations } = this.state;
+      this.setState({
+        calculatingResult: true
+      });
+      const worker = new ResultWorker();
```

Comment:
Current code has memory leak in many weapon and change often.
and worker does not promise the order of receive results.

1. store `worker` instance to global (property of `document` or `window`)
  (no need to care lock or atomic here, it's thread-safe action in main thread)
2. kill previous worker by invoke terminate method
  https://developer.mozilla.org/en-US/docs/Web/API/Worker/terminate
3. spawn new worker to global place. (see step 1)

This is not enough efficient solution, terminate() kills worker, we can't reuse the previous worker resource.
it's a limitation by the design of the worker. at least I expect it resolves memory leak.